python/tank/util/metrics.py

```diff
@@ -538,7 +538,7 @@ def _dispatch_to_endpoint(self, metrics):
             "auth_args": {"session_token": sg_connection.get_session_token()},
             "metrics": filtered_metrics_data,
         }
-        payload_json = json.dumps(payload)
+        payload_json = six.ensure_binary(json.dumps(payload))
```

Comment:
Are you saying that passing in the wrong data type raises an HTTPError? That's unfortunate!

Reply:
Actually it fails with a `TypeError`:
`TypeError: POST data should be bytes, an iterable of bytes, or a file object. It cannot be of type str.`